### ISSUE TITLE
[3.1.0-SNAPSHOT with data changes] ValueInstanciationException after creating a message containing a user @mention

### ISSUE THREAD

**Issue by Alex1304** [Discord4J/Discord4J#622]
**To Reproduce:** 

Create a message that contains at least one user @mention

**Expected Behavior:** 

The message sends, no error

**Actual Behavior:** 

The message sends, but when trying to read the response of the POST request an error occurs

```
com.fasterxml.jackson.databind.exc.ValueInstantiationException: Cannot construct instance of `discord4j.discordjson.json.ImmutableUserWithMemberData`, problem: Cannot build UserWithMemberData, some of required attributes are not set [member]
 at [Source: (byte[])"{"id": "690948607775604817", "type": 0, "content": "<@272872694473687041>, here is the profile of user **Alex1304**:", "channel_id": "527869943971053586", "author": {"id": "359406519227383818", "username": "UltimateGDBot Test Version", "avatar": null, "discriminator": "8299", "bot": true}, "attachments": [], "embeds": [{"type": "rich", "fields": [{"name": ":chart_with_upwards_trend:  Alex1304's stats", "value": "<:star:376445841885495316>  ` \u200c\u200c \u200c\u200c \u200c\u200c \u200c\u200c \u"[truncated 2411 bytes]; line: 1, column: 2728] (through reference chain: discord4j.discordjson.json.ImmutableMessageData$Json["mentions"]->java.util.ArrayList[0])
	at discord4j.rest.http.JacksonReaderStrategy.lambda$read$0(JacksonReaderStrategy.java:70)
	Suppressed: reactor.core.publisher.FluxOnAssembly$OnAssemblyException: 
Assembly trace from producer [reactor.core.publisher.MonoMapFuseable] :
	reactor.core.publisher.Mono.map(Mono.java:3044)
	discord4j.rest.http.JacksonReaderStrategy.read(JacksonReaderStrategy.java:65)
Error has been observed at the following site(s):
	|_           Mono.map ⇢ at discord4j.rest.http.JacksonReaderStrategy.read(JacksonReaderStrategy.java:65)
	|_      Mono.doOnNext ⇢ at discord4j.rest.http.JacksonReaderStrategy.read(JacksonReaderStrategy.java:75)
	|_         checkpoint ⇢ Body from POST /channels/527869943971053586/messages [ClientResponse]
	|_       Mono.flatMap ⇢ at discord4j.rest.http.client.ClientResponse.bodyToMono(ClientResponse.java:114)
	|_       Mono.flatMap ⇢ at discord4j.rest.request.DiscordWebResponse.bodyToMono(DiscordWebResponse.java:48)
	|_     Mono.publishOn ⇢ at discord4j.rest.request.DiscordWebResponse.bodyToMono(DiscordWebResponse.java:49)
	|_           Mono.map ⇢ at discord4j.core.object.entity.channel.BaseMessageChannel.createMessage(BaseMessageChannel.java:75)
```

**Version:** 
3.1.0-SNAPSHOT, includes recent data branch merge

**Comment by quanticc** [comment]
Thanks, I'll work on a fix

**Comment by quanticc** [comment]
Fixed via Discord4J/discord-json@ad46ba279a01e8b481b731ea078ee541fe5ff073
